app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -44,13 +43,18 @@ import org.kiwix.kiwixmobile.kiwixActivityComponent
 import org.kiwix.kiwixmobile.settings.KiwixSettingsActivity
 import org.kiwix.kiwixmobile.webserver.ZimHostActivity
 
-class KiwixMainActivity : CoreMainActivity() {
-  private lateinit var navController: NavController
-  private lateinit var appBarConfiguration: AppBarConfiguration
+const val PAGE_URL_KEY = "pageUrl"
+const val ZIM_FILE_URI_KEY = "zimFileUri"
 
+class KiwixMainActivity : CoreMainActivity() {
   private var actionMode: ActionMode? = null
 
-  val cachedComponent by lazy { kiwixActivityComponent }
+  override val cachedComponent by lazy { kiwixActivityComponent }
+  override val navController by lazy { findNavController(R.id.nav_host_fragment) }
+  override val drawerContainerLayout by lazy { navigation_container!! }
```

Comment:
these should not be nullable?

Reply:
oh, right. I thought it would figure that out by itself.